lukepolson/DicomAnnotator
# IAA-9 Change the logic where the pipeline can’t finish if files don’t convert properly (i.e. backend fails)

# This PR:
- Stiffens up the backend a bit
- Removes a response that caused an error when spawned python scripts failed
    - We listened for both 'error' and 'close' events, and sent a response for each of those events
    - When an 'error' event occurs, the 'close' event follows
    - Attempting to send 2 responses caused the backend to encounter a fatal error
- Includes https://github.com/cbarkr/DicomAnnotator/pull/3
- Includes https://github.com/cbarkr/DicomAnnotator/pull/4

# Notes: 
- See the linked PRs for more info on individual changes

# Testing: 
1. Pull the changes from this branch
2. Run `npm install`
3. Change the local input dir in `server.js` to wherever your input files are stored
4. Run `npm run build`
5. Run `npm run start`
6. Ensure that the app preexisting app functions still work as intended
7. Press the cancel button
8. Ensure that the logs (to the console) are output in JSON format, including log level, message, and timestamp
9. Make some annotations
10. Press the save button
11. Ensure that logs are created
12. Ensure that toast messages are created
13. Stop the app
14. Change the local input dir in `server.js` to a dir that doesn't exist
15. Follow the same steps as above (rebuild, restart) and ensure an error toast pops up
16. Check the logs and ensure that the backend does not close
